packages/@ember/-internals/extension-support/lib/data_adapter.js

```diff
@@ -92,6 +205,24 @@ export default EmberObject.extend({
    */
   acceptsModelName: true,
 
+  /**
+     Map from records arrays to RecordsWatcher instances
+
+     @private
+     @property recordsWatchers
+     @since 3.26.0
+   */
+  recordsWatchers: null,
```

Comment:
We don't need to initialize this to `null` on the prototype just to clobber that in the constructor

```suggestion
```